common-pg/src/main/scala/com/socrata/pg/analyzer2/SoQLFunctionSqlizerRedshift.scala

```diff
@@ -0,0 +1,729 @@
+package com.socrata.pg.analyzer2
+
+import com.socrata.prettyprint.prelude._
+import com.socrata.soql.analyzer2._
+import com.socrata.soql.collection.NonEmptySeq
+import com.socrata.soql.types._
+import com.socrata.soql.functions.SoQLFunctions._
+import com.socrata.soql.functions.{Function, MonomorphicFunction, SoQLTypeInfo}
+import SoQLFunctionSqlizerRedshift._
+
+class SoQLFunctionSqlizerRedshift[MT <: MetaTypes with ({ type ColumnType = SoQLType; type ColumnValue = SoQLValue })] extends FuncallSqlizer[MT] {
+  import SoQLTypeInfo.hasType
+
+  implicit class ExprInterpolator(sc: StringContext) {
+    def expr(args: Any*): OrdinaryFunctionSqlizer =
+      ofs { (f, runtimeArgs, ctx) =>
+        assert(args.length >= f.function.minArity)
+        assert(f.function.allParameters.startsWith(runtimeArgs.map(_.typ)))
+
+        val expandedRuntimeArgs = args.map(_.asInstanceOf[Int]).map(n => runtimeArgs.toList(n))
+
+      val sql = Doc(sc.s(expandedRuntimeArgs.map(_.compressed.sql): _*))
```

Comment:
..though I'm not sure I like this line here.  It smashes all the Docs together into a single String and loses all annotation information.

Reply:
I think I'd prefer
```scala
  implicit class ExprInterpolator(sc: StringContext) {
    def expr(args: Int*): OrdinaryFunctionSqlizer =
      ofs { (f, runtimeArgs, ctx) =>
        assert(args.length >= f.function.minArity)
        assert(f.function.allParameters.startsWith(runtimeArgs.map(_.typ)))

        val args = args.map(runtimeArgs(_).compressed.sql)
        val sql = (sc.parts.head +: (args, sc.parts.tail).zipped.map { (arg, lit) => arg ++ Doc(lit) }).hcat
        ExprSql(sql, f)
      }
    }
```

(two things here: `expr` now takes a variable number of Ints rather than Anys, and `sc.parts` will always have one more element than `args` so we can do that head/tail thing and then hcat it all together)

Reply:
what is hcat? This code does not compile for me.

Reply:
https://hackage.haskell.org/package/pretty-1.1.3.6/docs/Text-PrettyPrint.html#v:hcat

Reply:
ah, the compiler is confused. I think you want Doc(sc.parts.head)

Reply:
@rjmac Do we want to parenthesize the `Doc(lit)` ?